react/react-addons-test-utils.d.ts

```diff
@@ -6,7 +6,7 @@
 /// <reference path="react.d.ts" />
 
 declare namespace __React {
-    interface SyntheticEventData {
+    interface SyntheticEventData extends Event {
```

Comment:
I think that is the correct thing given @vsiao's statement. If the event attributes are optional, then they should also be marked as optional in the type definition